pkg/controller/job/job_controller_test.go

```diff
@@ -5254,34 +5254,50 @@ func TestFinalizerCleanup(t *testing.T) {
 
 	clientset := fake.NewSimpleClientset()
 	sharedInformers := informers.NewSharedInformerFactory(clientset, controller.NoResyncPeriodFunc())
+
 	manager, err := NewController(ctx, sharedInformers.Core().V1().Pods(), sharedInformers.Batch().V1().Jobs(), clientset)
 	if err != nil {
 		t.Fatalf("Error creating Job controller: %v", err)
 	}
 	manager.podStoreSynced = alwaysReady
 	manager.jobStoreSynced = alwaysReady
 
-	// Initialize the controller with 0 workers to make sure the
-	// pod finalizers are not removed by the "syncJob" function.
-	go manager.Run(ctx, 0)
-
 	// Start the Pod and Job informers.
 	sharedInformers.Start(ctx.Done())
 	sharedInformers.WaitForCacheSync(ctx.Done())
+	// Initialize the controller with 0 workers to make sure the
+	// pod finalizers are not removed by the "syncJob" function.
+	go manager.Run(ctx, 0)
 
 	// Create a simple Job
 	job := newJob(1, 1, 1, batch.NonIndexedCompletion)
 	job, err = clientset.BatchV1().Jobs(job.GetNamespace()).Create(ctx, job, metav1.CreateOptions{})
 	if err != nil {
-		t.Fatalf("Creating job: %v", err)
+		t.Fatalf("Could not create Job: %v", err)
+	}
+
+	// Await for the Job to appear in the jobLister to ensure so that Job Pod gets tracked correctly.
+	if err := wait.PollUntilContextTimeout(ctx, 100*time.Millisecond, wait.ForeverTestTimeout, false, func(ctx context.Context) (bool, error) {
+		job, _ := manager.jobLister.Jobs(job.GetNamespace()).Get(job.Name)
+		return job != nil, nil
+	}); err != nil {
+		t.Errorf("Waiting for Job object to appear in jobLister: %v", err)
```

Comment:
```suggestion
		t.Fatalf("Waiting for Job object to appear in jobLister: %v", err)
```
I think if we continue the test we can get misleading error messages.